flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1420,6 +1424,41 @@ private void timeoutSlot(AllocationID allocationId, UUID ticket) {
 		}
 	}
 
+	/**
+	 * Sync slot with job master.
+	 * The slots only exist in job master will be failed to job master.
+	 * The slots only exist in task executor will be released to resource manager.
```

Comment:
Rephrase: `Syncs the TaskExecutor's view on its allocated slots with the JobMaster's view. Slots which are no longer reported by the JobMaster are being freed. Slots which the JobMaster thinks it still owns but which are no longer allocated to it will be failed via {@link JobMasterGateway#failSlot}`.

Reply:
Done